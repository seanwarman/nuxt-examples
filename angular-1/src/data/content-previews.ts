import { RoomPlannerImages, SectionImages } from '../assets/images';

export interface ContentPreview {
  label: string;
  primaryImage: string;
  icon: string;
  body: string;
  slug: string;
  firstSubsection?: string;
  learnIcon: string;
  designIcon: string;
}

export const contentPreviews: ContentPreview[] = [
  {
    slug: 'camera-and-audio',
    label: 'Camera & Audio',
    firstSubsection: 'camera',
    primaryImage: SectionImages.cameraAndAudio.contentPreviewHero,
    icon: SectionImages.cameraAndAudio.icons.sectionIcon,
    designIcon: SectionImages.cameraAndAudio.icons.designIcon,
    learnIcon: SectionImages.cameraAndAudio.icons.learnIcon,
    body: `We've tuned and tweaked the hardware. Now, position it correctly to
     unlock the best performance.`,
  },
  {
    slug: 'displays-and-extras',
    label: 'Displays & Extras',
    firstSubsection: 'screen',
    primaryImage: SectionImages.displaysAndExtras.contentPreviewHero,
    icon: SectionImages.displaysAndExtras.icons.sectionIcon,
    designIcon: SectionImages.displaysAndExtras.icons.designIcon,
    learnIcon: SectionImages.displaysAndExtras.icons.learnIcon,
    body: `Correctly sizing displays is a key part of the conference experience.
      Plus, get some hints and tips on using the Series One Board 65 and Desk 27,
      developed in collaboration with Avocor,
      in your space.`,
  },
  {
    slug: 'room-layout',
    label: 'Room layout',
    firstSubsection: 'viewing-area',
    primaryImage: SectionImages.roomLayout.contentPreviewHero,
    icon: SectionImages.roomLayout.icons.sectionIcon,
    designIcon: SectionImages.roomLayout.icons.designIcon,
    learnIcon: SectionImages.roomLayout.icons.learnIcon,
    body: `Following a few spatial guidelines can make a huge difference for remote and in-room participants.`,
  },
  {
    slug: 'furniture-lighting',
    label: 'Furniture and lighting',
    firstSubsection: 'lighting',
    primaryImage: SectionImages.furnitureLighting.contentPreviewHero,
    icon: SectionImages.furnitureLighting.icons.sectionIcon,
    designIcon: SectionImages.furnitureLighting.icons.designIcon,
    learnIcon: SectionImages.furnitureLighting.icons.learnIcon,
    body: `Make your meeting room look and feel inviting with simple decorative and lighting tips. Appropriate lighting helps both local and remote participants have a great experience.`,
  },
  {
    slug: 'acoustics',
    label: 'Acoustics',
    firstSubsection: 'separation',
    primaryImage: SectionImages.acoustics.contentPreviewHero,
    icon: SectionImages.acoustics.icons.sectionIcon,
    designIcon: SectionImages.acoustics.icons.designIcon,
    learnIcon: SectionImages.acoustics.icons.learnIcon,
    body: `Poor acoustics is a common downfall of video conferencing spaces. The right allowance of acoustic finishes can help make speech clear and audible, and adequate isolation helps avoid interruptions.`,
  },
  {
    slug: 'services',
    label: 'Services',
    firstSubsection: 'air-handling',
    primaryImage: SectionImages.services.contentPreviewHero,
    icon: SectionImages.services.icons.sectionIcon,
    designIcon: SectionImages.services.icons.designIcon,
    learnIcon: SectionImages.services.icons.learnIcon,
    body: `Make sure power is provisioned for the right places, and ensure ventilation systems don't cause unwanted noise pickup on your microphones.`,
  },
  {
    slug: 'room-planner',
    label: 'Room Designer',
    primaryImage: RoomPlannerImages.contentPreviewHero,
    icon: RoomPlannerImages.icons.mainIcon,
    designIcon: SectionImages.services.icons.designIcon,
    learnIcon: SectionImages.services.icons.learnIcon,
    body: `Need to size up your room? All the best practices in this guide are used to help you pick the equipment and recommended room size for the occupancy you need.`,
  },
];
