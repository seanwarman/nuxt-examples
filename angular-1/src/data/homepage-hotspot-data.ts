export const HotspotData: Array<{
  name: string;
  text: string;
  color: string;
  defaultIcon: string;
  interimIcon: string;
  activeIcon: string;
  halo: string;
  x: number;
  y: number;
  z: number;
  cameraPosition: {
    x: number;
    y: number;
    z: number;
  };
  cameraRotation: {
    x: number;
    y: number;
    z: number;
  };
  modelsOn: Array<string>;
  modelsOff: Array<string>;
}> = [
  {
    name: 'camera-and-audio',
    text: 'Camera & Audio',
    color: '#e94233',
    defaultIcon: 'icon-hotspot-camera-audio-red-01-default@3x.png',
    interimIcon: 'icon-hotspot-camera-audio-red-01-interim@3x.png',
    activeIcon: 'icon-hotspot-camera-audio-red-01-active@3x.png',
    halo: 'icon-hotspot-camera-audio-red-01-active-halo@3x.png',
    x: -5,
    y: 113,
    z: -160,
    cameraPosition: {
      x: -100,
      y: 175,
      z: 350,
    },
    cameraRotation: {
      x: 100,
      y: 100,
      z: 0,
    },
    modelsOn: ['CameraAndAudio', 'defaultChairs', 'acousticDefault'],
    modelsOff: ['defaultCameraAndAudio'],
  },
  {
    name: 'displays-and-extras',
    text: 'Displays & Extras',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-displays-extras-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: -230,
    y: 150,
    z: 50,
    cameraPosition: {
      x: 100,
      y: 200,
      z: 500,
    },
    cameraRotation: {
      x: 50,
      y: 100,
      z: 0,
    },
    modelsOn: [
      'DisplaysAndExtras',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'room-layout',
    text: 'Room Layout',
    color: '#31a750',
    defaultIcon: 'icon-hotspot-room-layout-green-01-default@3x.png',
    interimIcon: 'icon-hotspot-room-layout-green-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-layout-green-01-active@3x.png',
    halo: 'icon-hotspot-room-layout-green-01-active-halo@3x.png',
    x: 90,
    y: 20,
    z: 200,
    cameraPosition: {
      x: 0,
      y: 280,
      z: 800,
    },
    cameraRotation: {
      x: 150,
      y: 100,
      z: 100,
    },
    modelsOn: [
      'RoomLayout',
      'defaultChairs',
      'defaultCameraAndAudioSetup',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'furniture-lighting',
    text: 'Furniture & Lighting',
    color: '#fbbc05',
    defaultIcon: 'icon-hotspot-furniture-lighting-yellow-01-default@3x.png',
    interimIcon: 'icon-hotspot-furniture-lighting-yellow-01-interim@3x.png',
    activeIcon: 'icon-hotspot-furniture-lighting-yellow-01-active@3x.png',
    halo: 'icon-hotspot-furniture-lighting-yellow-01-active-halo@3x.png',
    x: 110,
    y: 70,
    z: 80,
    cameraPosition: {
      x: 400,
      y: 200,
      z: 150,
    },
    cameraRotation: {
      x: 0,
      y: 70,
      z: -100,
    },
    modelsOn: [
      'FurnitureAndLighting',
      'defaultCameraAndAudioSetup',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: ['defaultChairs'],
  },
  {
    name: 'acoustics',
    text: 'Acoustics',
    color: '#3d85f2',
    defaultIcon: 'icon-hotspot-acoustics-blue-01-default@3x.png',
    interimIcon: 'icon-hotspot-acoustics-blue-01-interim@3x.png',
    activeIcon: 'icon-hotspot-acoustics-blue-01-active@3x.png',
    halo: 'icon-hotspot-acoustics-blue-01-active-halo@3x.png',
    x: -230,
    y: 200,
    z: 180,
    cameraPosition: {
      x: 300,
      y: 800,
      z: -150,
    },
    cameraRotation: {
      x: -150,
      y: 0,
      z: -100,
    },
    modelsOn: [
      'Acoustics',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticPanel00',
      'acousticPanel01',
      'acousticPanel02',
      'acousticPanel03',
      'defaultCameraAndAudio',
    ],
    modelsOff: ['stripLight00', 'stripLight01', 'acousticDefault'],
  },
  {
    name: 'services',
    text: 'Services',
    color: '#681da7',
    defaultIcon: 'icon-hotspot-services-purple-01-default@3x.png',
    interimIcon: 'icon-hotspot-services-purple-01-interim@3x.png',
    activeIcon: 'icon-hotspot-services-purple-01-active@3x.png',
    halo: 'icon-hotspot-services-purple-01-active-halo@3x.png',
    x: 110,
    y: 115,
    z: -160,
    cameraPosition: {
      x: 300,
      y: 160,
      z: 150,
    },
    cameraRotation: {
      x: 0,
      y: 50,
      z: -250,
    },
    modelsOn: [
      'services',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
  {
    name: 'room-planner',
    text: 'Room Designer',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-designer-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: -10,
    y: 90,
    z: 0,
    cameraPosition: {
      x: -30,
      y: 150,
      z: 150,
    },
    cameraRotation: {
      x: 40,
      y: 60,
      z: 0,
    },
    modelsOn: [
      'Drawing',
      'defaultCameraAndAudioSetup',
      'defaultChairs',
      'acousticDefault',
      'defaultCameraAndAudio',
    ],
    modelsOff: [],
  },
];
