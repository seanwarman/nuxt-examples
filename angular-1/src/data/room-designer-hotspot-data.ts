export const HotspotData: Array<{
  name: string;
  text: string;
  color: string;
  defaultIcon: string;
  interimIcon: string;
  activeIcon: string;
  halo: string;
  x: number;
  y: number;
  z: number;
  camX: number;
  camY: number;
  camZ: number;
}> = [
  {
    name: 'room-layout',
    text: 'Room Layout',
    color: '#31a750',
    defaultIcon: 'icon-hotspot-room-layout-green-01-default@3x.png',
    interimIcon: 'icon-hotspot-room-layout-green-01-interim@3x.png',
    activeIcon: 'icon-hotspot-room-layout-green-01-active@3x.png',
    halo: 'icon-hotspot-room-layout-green-01-active-halo@3x.png',
    x: -110,
    y: 20,
    z: 100,
    camX: -800,
    camY: 800,
    camZ: 800,
  },
  {
    name: 'camera-and-audio',
    text: 'Camera & Audio',
    color: '#e94233',
    defaultIcon: 'icon-hotspot-camera-audio-red-01-default@3x.png',
    interimIcon: 'icon-hotspot-camera-audio-red-01-interim@3x.png',
    activeIcon: 'icon-hotspot-camera-audio-red-01-active@3x.png',
    halo: 'icon-hotspot-camera-audio-red-01-active-halo@3x.png',
    x: -10,
    y: 90,
    z: 0,
    camX: -800,
    camY: 250,
    camZ: 300,
  },
  {
    name: 'displays-and-extras',
    text: 'Displays & Extras',
    color: '#0d9cab',
    defaultIcon: 'icon-hotspot-displays-extras-teal-01-default@3x.png',
    interimIcon: 'icon-hotspot-displays-extras-teal-01-interim@3x.png',
    activeIcon: 'icon-hotspot-displays-extras-teal-01-active@3x.png',
    halo: 'icon-hotspot-room-designer-teal-01-active-halo@3x.png',
    x: 0,
    y: 110,
    z: -160,
    camX: -1200,
    camY: 500,
    camZ: 600,
  },
  {
    name: 'services',
    text: 'Services',
    color: '#681da7',
    defaultIcon: 'icon-hotspot-services-purple-01-default@3x.png',
    interimIcon: 'icon-hotspot-services-purple-01-interim@3x.png',
    activeIcon: 'icon-hotspot-services-purple-01-active@3x.png',
    halo: 'icon-hotspot-services-purple-01-active-halo@3x.png',
    x: 120,
    y: 200,
    z: -160,
    camX: -120,
    camY: 250,
    camZ: 300,
  },
  {
    name: 'furniture-lighting',
    text: 'Furniture & Lighting',
    color: '#fbbc05',
    defaultIcon: 'icon-hotspot-furniture-lighting-yellow-01-default@3x.png',
    interimIcon: 'icon-hotspot-furniture-lighting-yellow-01-interim@3x.png',
    activeIcon: 'icon-hotspot-furniture-lighting-yellow-01-active@3x.png',
    halo: 'icon-hotspot-furniture-lighting-yellow-01-active-halo@3x.png',
    x: -70,
    y: 265,
    z: 20,
    camX: -120,
    camY: 300,
    camZ: 400,
  },
  {
    name: 'acoustics',
    text: 'Acoustics',
    color: '#3d85f2',
    defaultIcon: 'icon-hotspot-acoustics-blue-01-default@3x.png',
    interimIcon: 'icon-hotspot-acoustics-blue-01-interim@3x.png',
    activeIcon: 'icon-hotspot-acoustics-blue-01-active@3x.png',
    halo: 'icon-hotspot-acoustics-blue-01-active-halo@3x.png',
    x: -210,
    y: 190,
    z: -160,
    camX: -120,
    camY: 220,
    camZ: 580,
  },
  // {
  //   name: 'room-planner',
  //   text: 'Room Planner',
  //   color: '#0d9cab',
  //   defaultIcon: 'icon-hotspot-teal-01@3x.png',
  //   activeIcon: 'icon-room-designer-white-teal-circle-01@3x.png',
  //   halo: 'teal.png',
  //   x: -10,
  //   y: 150,
  //   z: -160,
  //   camX: 0,
  //   camY: 200,
  //   camZ: 200,
  // },
];
