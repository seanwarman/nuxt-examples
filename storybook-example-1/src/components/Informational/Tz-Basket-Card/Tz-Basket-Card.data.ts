export const exampleData = {
  id: 244988,
  title: 'Specialized Rockhopper Sport 27.5" Mountain Bike 2022',
  link: {
    url: '/.specialized-rockhopper-sport-27-5-mountain-bike-2022-hardtail-mtb_244988.htm?829688'
  },
  image: {
    src: '/244988/244988-829747_1_supersize.jpg',
    alt: ''
  },
  options: [
    {
      name: 'Colour',
      value: 'Gloss Blaze / Ice Papaya'
    },
    {
      name: 'Size',
      value: 'M'
    }
  ],
  price: {
    list: 575,
    rrp: 575
  },
  quantity: 1,
  stockQuantity: 5
};
