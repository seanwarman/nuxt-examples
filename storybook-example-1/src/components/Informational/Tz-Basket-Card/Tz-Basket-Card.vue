<template>
  <div>
    <div class="flex gap-9 bg-gray-50 p-[18px] sm:p-8 sm:pl-11 lg:gap-24">
      <div>
        <img
          class="pt-4.5 aspect-[3/2] w-[89px] object-contain mix-blend-multiply sm:w-[153px] sm:pt-0"
          :src="image.src"
          :alt="image.alt"
        />
      </div>
      <div class="flex-1">
        <div class="flex flex-col gap-2 lg:gap-7">
          <span class="text-base font-semibold leading-5 line-clamp-2">
            <tz-link :link="link">
              {{ title }}
            </tz-link>
          </span>
          <div class="flex flex-col gap-3">
            <ul v-if="options?.length" class="text-sm font-semibold leading-5 text-[#878787]">
              <li v-for="(option, k) in options" :key="k" class="line-clamp-2">
                {{ option.name }}: {{ option.value }}
              </li>
            </ul>
            <div class="flex flex-col gap-0">
              <span v-if="price" class="font-bold">
                {{ handlePrices(price.list) }}
              </span>
            </div>
            <tz-select
              :options="quantityOptions"
              :default-state="quatitySelection"
              @selected-option="onQuantityChange"
            />
            <div class="flex items-end justify-between">
              <button
                type="button"
                role="button"
                class="bg pb-[3px] text-xs font-bold uppercase tracking-widest text-gray-500 hover:underline focus:text-black focus:underline focus:outline-0"
                @click="onRemoveClick"
              >
                Remove
              </button>
              <div class="flex flex-col text-right">
                <span class="mb-0 text-base font-bold">
                  <span class="sr-only">Line price:</span> {{ handlePrices(linePrice) }}
                </span>
                <span
                  v-if="price.list < price.rrp"
                  class="m-0 pb-0.5 text-xs font-normal leading-normal tracking-normal text-gray-500"
                >
                  RRP: <span class="line-through">{{ handlePrices(lineRrp) }}</span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import type { Image, Link } from '@/components/Informational/types';
import { handlePrices } from '@/composables/numbers';
import TzLink from '@/components/Action/Tz-Link/Tz-Link.vue';
import TzSelect from '@/components/Action/Tz-Select/Tz-Select.vue';
import { defineProps, ref, computed, onMounted } from 'vue';
import type { Option } from '../../Input/types';

export interface Props {
  id: number;
  link: Link;
  image: Image;
  title: string;
  options?: Option[];
  price: {
    list: number;
    rrp: number;
  };
  stockQuantity: number;
  quantity: number;
}

const props = defineProps<Props>();
const quantity = ref(props.quantity);

const linePrice = computed(() => props.price.list * quantity.value);
const lineRrp = computed(() => props.price.rrp * quantity.value);
const quantityOptions = computed(() =>
  Array.from({ length: Math.min(20, props.stockQuantity) }, (_, index) => index + 1).reduce(
    (arr: Option[], value: number) => [
      ...arr,
      {
        name: value.toString(),
        value
      }
    ],
    []
  )
);
const quatitySelection = computed(() => {
  return { name: props.quantity.toString(), value: props.quantity };
});

const emit = defineEmits<{
  (eventName: 'remove', basketItemId: number): void;
  (eventName: 'changeQuantity', basketItemId: number, quantity: number): void;
}>();

const onRemoveClick = () => {
  emit('remove', props.id);
};
const onQuantityChange = (option: Option) => {
  quantity.value = Number(option.value);
  emit('changeQuantity', props.id, parseInt(option.value as string));
};
</script>
