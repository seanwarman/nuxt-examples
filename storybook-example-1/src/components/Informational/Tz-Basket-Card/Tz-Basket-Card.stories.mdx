import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs';
import { action } from '@storybook/addon-actions';

import TzBasketCard from './Tz-Basket-Card.vue';
import { exampleData } from './Tz-Basket-Card.data';
import { MobileViewports } from '@/helpers/viewports';

<Meta title="components/Informational/Basket Card" component={TzBasketCard} />

export const Template = (args) => ({
  components: { TzBasketCard },
  setup() {
    return { args };
  },
  template: '<tz-basket-card v-bind="args" @remove="remove" @changeQuantity="changeQuantity"/>',
  methods: {
    remove: action('remove'),
    changeQuantity: action('changeQuantity')
  }
});

# Basket Card

## Full RRP

<Canvas withSource>
  <Story
    name="Default"
    args={exampleData}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## Reduced price

<Canvas withSource>
  <Story
    name="Reduced Price"
    args={{ ...exampleData, ...{ price: { list: 550, rrp: 575 } } }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## No colour

<Canvas withSource>
  <Story
    name="No Colour"
    args={{
      ...exampleData,
      ...{
        options: [
          {
            name: 'Size',
            value: 'M'
          }
        ]
      }
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## No size

<Canvas withSource>
  <Story
    name="No Size"
    args={{
      ...exampleData,
      ...{
        options: [
          {
            name: 'Colour',
            value: 'Gloss Blaze / Ice Papaya'
          }
        ]
      }
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## No size or color

<Canvas withSource>
  <Story
    name="No Size or Color"
    args={{
      ...exampleData,
      ...{
        options: []
      }
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## Very long colour

<Canvas withSource>
  <Story
    name="Very long colour"
    args={{
      ...exampleData,
      ...{
        options: [
          {
            name: 'Colour',
            value: 'Gloss Blaze / Ice Papaya / Acid Lime / Dark Coal / Sky Ground'
          },
          {
            name: 'Size',
            value: 'M'
          }
        ]
      }
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## Portrait Image

The image should be of fixed aspect ratio so a portrait image should be contained within that aspect ratio.

<Canvas withSource>
  <Story
    name="Portrait image"
    args={{
      ...exampleData,
      ...{
        image: {
          src: '109364-405055_1_Supersize.jpg',
          alt: ''
        },
        title: 'Cane Creek Valt Steel Spring For Double Barrel'
      }
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## Multiple Units

<Canvas withSource>
  <Story
    name="Multiple units"
    args={{
      ...exampleData,
      ...{
        price: { list: 550, rrp: 575 },
        quantity: 3
      }
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## Multiple Units > 20

Quantity selector to max to 20 items

<Canvas withSource>
  <Story
    name="Multiple units > 20"
    args={{
      ...exampleData,
      ...{
        price: { list: 550, rrp: 575 },
        quantity: 4,
        stockQuantity: 30
      }
    }}
    parameters={{
      chromatic: { viewports: MobileViewports }
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>
