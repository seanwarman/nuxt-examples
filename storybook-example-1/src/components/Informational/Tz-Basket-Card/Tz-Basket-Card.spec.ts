/**
 * @vitest-environment jsdom
 */

import { VueWrapper, mount } from '@vue/test-utils';
import TzBasketCard from './Tz-Basket-Card.vue';
import { expect, test, describe, beforeEach, it } from 'vitest';

import { exampleData } from './Tz-Basket-Card.data';

let wrapper: VueWrapper;

describe('Button setup', () => {
  beforeEach(() => {
    wrapper = mount(TzBasketCard, {
      props: exampleData
    });
  });
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('create correct amount of quantity options', () => {
    const select: HTMLSelectElement = wrapper.find('select').element;
    expect(select.options.length).toBe(exampleData.stockQuantity);
  });
  it('sets a maximum of 20 quantity options', async () => {
    wrapper = mount(TzBasketCard, {
      props: { ...exampleData, ...{ stockQuantity: 30 } }
    });
    await wrapper.vm.$nextTick();
    const select: HTMLSelectElement = wrapper.find('select').element;
    expect(select.options.length).toBe(20);
  });
  it('calculates total amount when quantity changed', async () => {
    const button = wrapper.find('button');
    button.trigger('click'); //click quantity select box
    await wrapper.vm.$nextTick();
    const item = wrapper.findAll('ul')[1].find('li:nth-child(2) a');
    item.trigger('click'); //click quantity option
    await wrapper.vm.$nextTick();
    const spans = wrapper.findAll('span');
    //check text
    expect(spans[spans.length - 2].element.textContent).contains((exampleData.price.list * 2).toLocaleString());
  });
});
