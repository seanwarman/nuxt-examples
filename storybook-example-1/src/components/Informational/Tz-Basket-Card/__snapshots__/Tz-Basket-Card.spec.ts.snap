// Vitest Snapshot v1

exports[`Button setup > renders correctly 1`] = `
<div>
  <div
    class="flex gap-9 bg-gray-50 p-[18px] sm:p-8 sm:pl-11 lg:gap-24"
  >
    <div>
      <img
        alt=""
        class="pt-4.5 aspect-[3/2] w-[89px] object-contain mix-blend-multiply sm:w-[153px] sm:pt-0"
        src="/244988/244988-829747_1_supersize.jpg"
      />
    </div>
    <div
      class="flex-1"
    >
      <div
        class="flex flex-col gap-2 lg:gap-7"
      >
        <span
          class="text-base font-semibold leading-5 line-clamp-2"
        >
          <a
            class="hover:underline focus:underline focus:outline-none"
            href="/.specialized-rockhopper-sport-27-5-mountain-bike-2022-hardtail-mtb_244988.htm?829688"
          >
            
            Specialized Rockhopper Sport 27.5" Mountain Bike 2022
            
          </a>
        </span>
        <div
          class="flex flex-col gap-3"
        >
          <ul
            class="text-sm font-semibold leading-5 text-[#878787]"
          >
            
            <li
              class="line-clamp-2"
            >
              Colour: Gloss Blaze / Ice Papaya
            </li>
            <li
              class="line-clamp-2"
            >
              Size: M
            </li>
            
          </ul>
          <div
            class="flex flex-col gap-0"
          >
            <span
              class="font-bold"
            >
              £575.00
            </span>
          </div>
          <div>
            <div
              class="tz-select-container relative inline w-full"
            >
              <!-- controls -->
              <button
                class="inline-flex w-full items-center justify-between gap-0 rounded-[5px] border p-1 focus:border-dark focus:ring-0 overflow-hidden focus-within:ring border-gray-400"
                data-test="tz-select-button"
                id="tzSelectButton"
                tabindex="0"
              >
                <!-- value content -->
                <div>
                  <span
                    class="block flex items-center whitespace-nowrap py-1 leading-3 px-3"
                    data-test="tz-select-icon"
                  >
                    <!--v-if-->
                     1
                  </span>
                </div>
                <!-- button section -->
                <div
                  class="border-l border-l-gray-400 px-2 flex items-center justify-center"
                  data-test="tz-select-chevron-wrapper"
                >
                  <div
                    class="transition-transform duration-300 rotate-90"
                  >
                    <svg
                      class="block w-5 text-primary scale-75"
                      fill="none"
                      viewBox="0 0 32 32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M19.775 16 4.555 7.629l2.89-5.258L24.26 11.62c3.454 1.9 3.454 6.863 0 8.762L7.447 29.63l-2.893-5.26L19.774 16Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                    <span
                      class="sr-only"
                    >
                       Select an option, you have selected 1
                    </span>
                  </div>
                </div>
              </button>
              <!-- options -->
              <transition-stub>
                <!--v-if-->
              </transition-stub>
              <!-- Phantom select - useful for forms -->
              <div
                class="hidden"
              >
                <label
                  for="dropdown"
                />
                <select
                  id="dropdown"
                  name="dropdown"
                >
                  
                  <option
                    selected="selected"
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  
                </select>
              </div>
            </div>
          </div>
          <div
            class="flex items-end justify-between"
          >
            <button
              class="bg pb-[3px] text-xs font-bold uppercase tracking-widest text-gray-500 hover:underline focus:text-black focus:underline focus:outline-0"
              role="button"
              type="button"
            >
               Remove 
            </button>
            <div
              class="flex flex-col text-right"
            >
              <span
                class="mb-0 text-base font-bold"
              >
                <span
                  class="sr-only"
                >
                  Line price:
                </span>
                 £575.00
              </span>
              <!--v-if-->
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
